drivers/shared/executor/executor.go

```diff
@@ -303,8 +306,30 @@ func (e *UniversalExecutor) Launch(command *ExecCommand) (*ProcessState, error)
 	e.childCmd.Env = e.commandCfg.Env
 
 	// Start the process
-	if err := e.childCmd.Start(); err != nil {
-		return nil, fmt.Errorf("failed to start command path=%q --- args=%q: %v", path, e.childCmd.Args, err)
+	if command.NetworkIsolation != nil && command.NetworkIsolation.Path != "" {
+		// Lock to the thread we're changing the network namespace of
+		runtime.LockOSThread()
+		netns, err := ns.GetNS(command.NetworkIsolation.Path)
+		if err != nil {
+			return nil, err
```

Comment:
nitpick: might be nice to wrap this error to get better logs, e.g.
```suggestion
			return nil, fmt.Errof("failed to get ns %s: %v", command.NetworkIsolation.Path, err)
```